### PR TITLE
Simpler Kernel API

### PR DESCRIPTION
**Pull request by DonJayamanne**
Reduces the npm packages required by extension authors.
ws & rxjs packages are no longer required to use the kernel API
Also makes it easier to consume the Socket events (users won't have to bind/unbind/bind again, it will be handled internally when kernels restarts)

### CODE REVIEW

File: src/client/api/kernelApi.ts [hunk 1]
```diff
@@ -184,3 +194,72 @@ class JupyterKernelService implements IExportedKernelService {
...
     }
 }
+
+/**
+ * Helper class to wrap the IKernelSocket.
+ * This way users of the API will not need to unbind/rebind the hooks and
+ * listen to changes in the observable.
+ * Also prevents the need for users of the API to depend on rxjs.
+ */
+class KernelSocketWrapper implements IKernelSocket {
```
Comment (DonJayamanne) [inline src/client/api/kernelApi.ts:204]: See PR comments

Comment (DonJayamanne) [reply src/client/api/kernelApi.ts:204]: Also class comments makes it easier to understand the purpose

File: src/client/api/extension.d.ts [hunk 2]
```diff
@@ -5,8 +5,11 @@
 import { CancellationToken, Event, NotebookDocument } from 'vscode';
 import type { Kernel } from '@jupyterlab/services/lib/kernel';
 import type { Session } from '@jupyterlab/services';
-import { Observable } from 'rxjs/Observable';
-import type { Data as WebSocketData } from 'ws';
```
Comment (DonJayamanne) [inline src/client/api/extension.d.ts:9]: Users of this API will not need ws & rxjs anymore.

**Review by rchiodo — APPROVED** [review]

**Comment by rchiodo** [comment]
Perf test failing can be ignored. You just need to merge from main.

**Pull request merged**